pkg/operator/configobserver/proxy/observe_proxy.go

```diff
@@ -60,6 +60,9 @@ func (f *observeProxyFlags) ObserveProxyConfig(genericListers configobserver.Lis
 		if err := unstructured.SetNestedStringMap(observedConfig, newProxyMap, f.configPath...); err != nil {
 			errs = append(errs, err)
 		}
+	} else if len(currentProxyMap) == 0 {
+		// both are unset
+		return existingConfig, errs
```

Comment:
why is this necessary? Because empty map vs. nil is not deep equal?

Reply:
exactly